### ISSUE TITLE
MIssing the possibility to provide custom templates for the managed files

### ISSUE THREAD

**Issue by alvagante** [echocat/puppet-graphite#153]
I need to provide custom templates for some of the module's managed files.
We can easily add parameters that allow users to provide their own templates (and, as default values have the currently used ones).
Rationale is better described here: http://www.example42.com/2014/10/29/reusability-features-every-module-should-have/ 

If you are interested in this I can prepare a PR.
My current alternative is to fork the module, which is something I'd definitively try to avoid.
Please let me know.

**Comment by dwerder** [comment]
Useful PRs are always ok. If it does not change the whole default behavior of the module.

**Comment by dwerder** [comment]
Cool blog.

**Comment by alvagante** [comment]
I looked at the code and the number of templates to "parametrize" is a bit more than I thought (18), adding parameters and documentation for each of them would take me some time that I have not right now. In the mean time I managed to achieve what I needed without modifying the module (this is my primary indication of the reusability of a module, so good work!), so I don't have such an urgency any more.
If you want to close the ticket, I have no objection.
+1 if you plan, sooner or later, to add parameters that allow customisation of templates, it's easy, fast, safe and backwards compatible.
